sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -915,15 +915,15 @@ case class MapObjects private(
   }
 
   private lazy val mapElements: scala.collection.Seq[_] => Any = customCollectionCls match {
-    case Some(cls) if classOf[WrappedArray[_]].isAssignableFrom(cls) =>
+    case Some(cls) if classOf[mutable.ArraySeq[_]].isAssignableFrom(cls) =>
```

Comment:
isn't it a breaking change? This is not an internal usage. This is to handle the user provided objects within the encoder.

Reply:
```scala
  @deprecated("Use ArraySeq instead of WrappedArray; it can represent both, boxed and unboxed arrays", "2.13.0")
  type WrappedArray[X] = ArraySeq[X]
  @deprecated("Use ArraySeq instead of WrappedArray; it can represent both, boxed and unboxed arrays", "2.13.0")
  val WrappedArray = ArraySeq
```

In Scala 2.13, `WrappedArray` is just an alias for `scala.collection.mutable.ArraySeq`, and it's not an independent type. Did you encounter any bad cases?

Reply:
```
Welcome to Scala 2.13.12 (OpenJDK 64-Bit Server VM, Java 17.0.9).
Type in expressions for evaluation. Or try :help.

scala> classOf[scala.collection.mutable.WrappedArray[_]]
              ^
       warning: type WrappedArray in package mutable is deprecated (since 2.13.0): Use ArraySeq instead of WrappedArray; it can represent both, boxed and unboxed arrays
val res0: Class[collection.mutable.WrappedArray[_]] = class scala.collection.mutable.ArraySeq

scala> classOf[scala.collection.mutable.ArraySeq[_]]
val res1: Class[scala.collection.mutable.ArraySeq[_]] = class scala.collection.mutable.ArraySeq
```

Using the Scala 2.13 shell, `classOf[scala.collection.mutable.WrappedArray[_]]` and `classOf[scala.collection.mutable.ArraySeq[_]]` both return `class scala.collection.mutable.ArraySeq`, indicating that there is no actual `WrappedArray` type in Scala 2.13.

Reply:
Ah sorry, I misread the scala source code of 2.12

Reply:
OK